assets/js/modules/analytics/setup.js

```diff
@@ -320,7 +330,8 @@ class AnalyticsSetup extends Component {
 				selectedProfile,
 				properties: [ chooseAccount ],
 				profiles: [ chooseAccount ],
-				existingTag: responseData.existingTag ? responseData.existingTag.property[ 0 ].id : false,
+				existingTag: existingTagData ? existingTagData.property : false,
+				useSnippet: ( ! existingTagData && ! onSettingsPage ) ? true : useSnippet,
```

Comment:
Shouldn't this be the following:
* if not on settings page and there is no existing tag: true
* if not on settings page and there is an existing tag: false
* if on settings page: original `useSnippet` value

Reply:
@felixarntz `useSnippet` is false by default when setting up Analytics module. 
We can specify this also when declaring state `useSnippet: useSnippet || false` so it's clear. 
I believe the logic right now `( ! existingTagData && ! onSettingsPage ) ? true : useSnippet` and being `useSnippet` false by default covers the cases you described above. 
Does this make sense?

Reply:
In theory this logic is sufficient, but it's error-prone as is. The logic itself does not cover for:
> if not on settings page and there is an existing tag: false

Sure, `useSnippet` will probably be false, but that very much depends on the rest of the logic, so in terms of defensive coding, we should ensure the above condition works.

Reply:
@felixarntz https://github.com/google/site-kit-wp/pull/434/commits/ec173196572d8ee95839f2a722a1159303c566d6